src/parse.rs

```diff
@@ -14,13 +14,18 @@ use crate::alloc::vec::Vec;
 
 /// Trait for objects that can be deserialized from hex strings.
 pub trait FromHex: Sized {
+    /// Error type returned while parsing hex string.
+    type Error: From<HexToBytesError> + Sized + fmt::Debug + fmt::Display;
```

Comment:
We have these sorts of constraints in rust-miniscript. Without them you can't `.expect` on generic errors. (Though maybe this is okay for this library, because like `FromStr`, its operations are small and will typically only occur on concrete types.)

Reply:
Thanks, I removed the `Send`, `Debug`, and `Display`. Cannot work out how to use the `Into` suggestion?

Reply:
I don't know why `FromStr` and `TryFrom` don't require `Debug`, an error type seems pretty useless if it does not implement even `Debug`?

Reply:
There are some niche use cases for not requiring `Debug`, anyway it doesn't prevent you from calling `expect` because the generic can just add `Foo::Error: fmt::Debug` bound.